### ISSUE TITLE
[Epic] Options creation flow

### ISSUE THREAD

**Issue by pmcpinto** [woothemes/woocommerce#35659]
**Description:** Users can create and manage variations through a dedicated tab rather than selecting the product variations type through a dropdown.

**Issues**

* [x] [Top navigation tabs [MVP]#35706](https://github.com/woocommerce/woocommerce/issues/35706)
* [x] [Product variations data store#35771](https://github.com/woocommerce/woocommerce/issues/35771)
* [x] [Render product variations list#35772](https://github.com/woocommerce/woocommerce/issues/35772)
* [x] [Refactor attributes component to reuse for options#35770](https://github.com/woocommerce/woocommerce/issues/35770)
* [x] [Automatically generate variations from product options#35778](https://github.com/woocommerce/woocommerce/issues/35778)
* [x] [Disable unnecessary product tabs when variations are created#35779](https://github.com/woocommerce/woocommerce/issues/35779)

**Acceptance criteria:**
- [x] Create a new tab called "Options" after the "Shipping tab.
- [x] If an existing product has 0 local or global attributes with "Used for variations" enabled, display the Options empty state with the "add options" CTA:  BaoQ3PGPnhAiJCsJtxgej3-fi-102%3A1247&t=EFqwqGcibTmy71Eq-0
- [ ] If an existing product has at least 1 local or global attribute with "Used for variations" enabled, display the Options section with a list including the available attributes + values. This section will behave exactly as the Attributes section: BaoQ3PGPnhAiJCsJtxgej3-fi-102%3A4612&t=EFqwqGcibTmy71Eq-0
- [x] If the user clicks "Add options", we'll open the "Add product options" modal, and the entire flow reuses the attributes creation flow: BaoQ3PGPnhAiJCsJtxgej3-fi-102%3A1763&t=EFqwqGcibTmy71Eq-0
- [ ] The option dropdown field is immediately in focus and shows the first 10 options (product/local attributes with ‘Used for variations’ enabled).
- [ ] Each product option should be saved as a global attribute used for variations. By default, all attributes are created globally. We inform the user about this in the add attributes modal.
- [ ] Each product option can be edited through a modal, where it will be possible to change the option name, add/remove values, define a default value, and define if it should be displayed to customers  (local vs global): BaoQ3PGPnhAiJCsJtxgej3-fi-102%3A2377&t=EFqwqGcibTmy71Eq-0
- [ ] The first variation in each option is selected by default on the product page. Users can remove it by selecting `None` from the `Default value (optional` dropdown in the option edit modal (in the current experience, we don't have a default value selected, which causes some quirky behaviors). When `None` is selected for all options, no variation will be selected when the user enters the page.
- [ ] Each option should have the box 'Show in product details' unchecked by default.
- [ ] Local attributes are marked with a small info icon. It is displayed in the Attributes and Options lists. When the user hovers over the icon, they will see a tooltip explaining that a local attribute will not be used for filtering, etc.
- [ ] Users can create several global attributes with the same name. We’d generate a unique slug for each based on its name. For example, the slugs for Color and Color would be ‘color’ and ‘color-2’.
- [x] Users need to go to the Attributes screen, to edit the attribute advanced details such as slug or archive.
- [ ] The variation list should be updated if the user adds or removes values. For example, if the user adds a new value to a "Size" option, we should auto-create all the new variations based on the new size value.
- [x] When at least one option is created, we should automatically create variations from all options. The product should be transformed into a variable product.
- [x] The variations should be displayed in a Variations section. We show a spinner while the variations are being generated (same that we use in the images section): BaoQ3PGPnhAiJCsJtxgej3-fi-103%3A42557&t=EFqwqGcibTmy71Eq-0
- [x] When the variations are created, we fade out/disable the Pricing, Inventory, and Shipping tabs of the parent product and display an overlay informing users about what was created: BaoQ3PGPnhAiJCsJtxgej3-fi-102%3A3315&t=EFqwqGcibTmy71Eq-0
- [x] When the user hovers over one of the disabled tabs, we show a tooltip reading: `Manage individual variation details in the Options tab.`
- [ ] If the users add an attribute in the attributes section, it will have the variations flag disabled. If the attribute is added in the options section, the flag will be enabled. There will be no way to enable or disable the flag once the attribute is added. To do that, users need to remove an attribute and add it in a different section: Attributes -> Options or Options -> Attributes.
- [x] If an attribute is added in one section, it can’t be selected in another. Users have to remove it from one section and add it to another. We would detect when an attribute was already added somewhere else and let users move it from one section to another.
- [ ] When users add a global attribute to Attributes, we automatically add its first value. When they add it to Options, we add all of its values.
- [x] Users can select global attributes in both places. If they create new values, they’re added to the attribute but not selected in the products it’s assigned to.
- [ ] Attributes and options can be sorted separately. Options are, by default, hidden from the product description. If the user enables the visibility toggle, options will be displayed below regular attributes.
- [ ] Changing the options order will be reflected at the top of the page and in the product description.
- [ ] If a product has 7 attributes: 2 options and 5 regular attributes, they can move n2 above n1, but not n7 above n2. They are managed in separate lists, even though they’re displayed in the same area in the product description.
- [ ] When users switch from the new experience back to the classic one, attributes from both sections will be displayed in a single list. We’ll respect the sorting order and place regular attributes below attributes used for variations. Users will be able to move them up and down however they want.

The variations list taxonomy/behavior is included in another issue.

**Event tracking:**
All user interactions should be tracked

**Comment by joshuatf** [comment]
>'Used for variations' box checked

This does not yet exist on attributes.  Should we add this before tackling the options?

>If the user clicks "Add options", we'll open the "Add product options" modal and the entire flow reuses the attributes creation flow

Assuming we implement the above variations checkbox, I assume this checkbox will not show in this flow, right?

>When the variations, we fade out/disable the Pricing, Inventory, and Shipping tabs

This is meant to say after we create a variation, right?

cc @pmcpinto @jarekmorawski

**Comment by jarekmorawski** [comment]
> This does not yet exist on attributes. Should we add this before tackling the options?

It exists if you select a variable product type and go to any attribute details. It doesn't show up for other product types, though.

> Assuming we implement the above variations checkbox, I assume this checkbox will not show in this flow, right?

Yes. The checkbox is downgraded to the role of an ID that allows us to distinguish regular attributes from variation-attributes (options) on the client side. If an attribute has this box checked, it's an option and appears in the `Options` tab. If it doesn't, it's a regular attribute and should show up in `General -> Attributes`.

> This is meant to say after we create a variation, right?

Correct! I've fixed the AC.

**Comment by joshuatf** [comment]
>It exists if you select a variable product type and go to any attribute details. It doesn't show up for other product types, though.

@jarekmorawski I meant in the new product management experience.  I don't think we have the option for variable products there or the checkboxes in attributes AFAIK.  Please let me know if I'm wrong.

**Comment by jarekmorawski** [comment]
> I don't think we have the option for variable products there or the checkboxes in attributes AFAIK. Please let me know if I'm wrong.

That's correct. In the new experience, all attributes created in `Options` will have the box checked (again, only on the server side because it won't be visible to users), while those added in `General` will have the box unchecked.

**Comment by joshuatf** [comment]
Ah I see.  There won't be a visual checkbox element, but the flag for marking an item as used for variations will be set to `true` for options and `false` for attributes, is that right?

**Comment by jarekmorawski** [comment]
That's correct. 👍 Those flags will also help us assign existing attributes to the right UI areas. Attributes with the flag set to `true` will go to Options and those set to `false` will go to Attributes.

**Comment by mattsherman** [comment]
> That's correct. 👍 Those flags will also help us assign existing attributes to the right UI areas. Attributes with the flag set to true will go to Options and those set to false will go to Attributes.

@jarekmorawski @louwie17 @joshuatf 

A few questions...

1. Should the "Attributes" list only show attributes with the `variation` flag set to `false`? In other words, *not* show any attributes being used for variations? Or, should all attributes for the product be shown in Attributes?
2. In the "Add product options" modal, should local attributes be listed to select from, as global attributes are?
3. In the "Add product options" modal, if a global (or local) attribute is selected that is already in "Attributes", I assume we should default the terms to the ones already selected in "Attributes"? And, if the terms are modified for an attribute in "Options", it should also be changed in "Attributes" (the way it works under the hood, data-wise, I see no way it could work any way other than this!)?

Based on the answer to (1) above, the following questions may or may not be valid...

4. If an attribute is removed from the "Options" list, should it also be removed from the "Attributes" list? Or, should only the `variation` flag for it be set to `false`?

**Comment by jarekmorawski** [comment]
> Should the "Attributes" list only show attributes with the variation flag set to false? In other words, not show any attributes being used for variations? Or, should all attributes for the product be shown in Attributes?

Correct. Attributes with the variation flag set to true should appear in `Options` (for both newly created and existing products).

> In the "Add product options" modal, should local attributes be listed to select from, as global attributes are?

No, it should only list global attributes with the variation flag enabled.

> And, if the terms are modified for an attribute in "Options", it should also be changed in "Attributes" (the way it works under the hood, data-wise, I see no way it could work any way other than this!)?

In the Add product options modal, I should only see attributes with the variation flag enabled, so there will be no overlap between Attributes and Options. An attribute can't exist in both places. The variation flag acts as a filter. If it's enabled, the attribute is an option. If it isn't, it shows up in Attributes. If I modify terms in Options, they will not affect the regular attributes because they're separate objects in the UI.

You cannot add an attribute with the flag enabled to Attributes. You can't also add an Option with the flag disabled.

> If an attribute is removed from the "Options" list, should it also be removed from the "Attributes" list? Or, should only the variation flag for it be set to false?

In this case, it should be removed from Options and hidden from the UI (when removed, it'll still be an attribute with the variation flag enabled and the only way to change it is through the global `Attributes` settings page). There's no connection between Options and Attributes (other than a technical one). Users can manipulate them independently.

**Comment by mattsherman** [comment]
Thanks @jarekmorawski! Some follow up questions...

> Correct. Attributes with the variation flag set to true should appear in Options (for both newly created and existing products).

And conversely, attributes with the variation flag set to true should not appear in Attributes, correct? (That is how I read the rest of your response, but I want to make sure I am crystal clear about this!)

> In this case, it should be removed from Options and hidden from the UI (when removed, it'll still be an attribute with the variation flag enabled and the only way to change it is through the global Attributes settings page).

There is no `variation` flag on the global attributes themselves.

From the AC:

> Each option should have the boxes 'Show in product details' and 'Use for filtering' checked by default.

It sounds from your comments above that we actually never should display the `Used for filters` checkbox in the new product experience UI. Is this correct?

<img width="593" alt="Screenshot 2022-12-08 at 13 35 37" src="https://user-images.githubusercontent.com/2098816/206539962-0bb61093-f9db-43cb-8c0f-eab4674a3dff.png">

(This is in the current new product experience)

> There's no connection between Options and Attributes (other than a technical one). Users can manipulate them independently.

Well, mostly, with two limitations...

- A global attribute can not be in both Options and Attributes... it is either/or. That has been implied, but just stating explicitly to make sure we are on the same page!
- A local attribute with the same name can not be in both Options and Attributes. (The current old UI actually lets you add another local attribute with the same name but then silently just keeps the newer one!)

So, if an attribute is currently selected in one list, it should not be able to be selected in the other list.

**_Side note: Our use of the term `Option` in the UI is confusing from a development standpoint, because the terms (global terminology)/values (local terminology) for an attribute are labeled as `options` in the REST API!_**

**Comment by mattsherman** [comment]
> Well, mostly, with two limitations...
> - A global attribute can not be in both Options and Attributes... it is either/or. That has been implied, but just stating explicitly to make sure we are on the same page!
> - A local attribute with the same name can not be in both Options and Attributes. (The current old UI actually lets you add another local attribute with the same name but then silently just keeps the newer one!)
So, if an attribute is currently selected in one list, it should not be able to be selected in the other list.

@jarekmorawski How do we want to handle this, from a UX perspective? It is easy to just not make them selectable from the other list, but I question if it will be clear to the user why attributes that are in the other list can't be selected.

**Comment by mattsherman** [comment]
@jarekmorawski Another big question...

Since we are going to show attributes in two separate sortable lists (Options and Attributes), how will the user sort all the attributes in a single ordering? That is what the current experience offers and what the front-end uses.

**Comment by jarekmorawski** [comment]
> It is easy to just not make them selectable from the other list, but I question if it will be clear to the user why attributes that are in the other list can't be selected.

We'd just hide them from the list. The next step is to create a global Options screen, just like the one for attributes. This will allow users to manage global options without going into each product individually.

> how will the user sort all the attributes in a single ordering?

How are they sorted now? They'll be two separate lists, so we can use the existing sorting system.

**Comment by jarekmorawski** [comment]
> And conversely, attributes with the variation flag set to true should not appear in Attributes, correct? (That is how I read the rest of your response, but I want to make sure I am crystal clear about this!)

That's correct!

> There is no variation flag on the global attributes themselves.

We might want to add it at some point in the future, but only if and when we decide to create a global `Products > Options` screen.

> It sounds from your comments above that we actually never should display the Used for filters checkbox in the new product experience UI. Is this correct?

Glad you asked because the second box (`Used for filters`) doesn't exist in the current experience and we're 100% if it needs to be added, though. The other one, `Visible to customers`, should be visible in the `Edit attribute` and `Edit option `modals (it's the equivalent of `Visible on the product page` box in the current UX, which shows up in attributes in both simple and variable products).

> A global attribute can not be in both Options and Attributes... it is either/or. That has been implied, but just stating explicitly to make sure we are on the same page!

In the early stages, we won't have any global options and global attributes won't show up in the Options dropdown because they wouldn't have the variation flag enabled. In the next stage, when we add a global Options screen, users will be able to create and manage global options that will be selectable in the dropdown for all products.

Tl;dr: You shouldn't be able to select a global attribute in Options.

> A local attribute with the same name can not be in both Options and Attributes. (The current old UI actually lets you add another local attribute with the same name but then silently just keeps the newer one!)

Oh, really! Thanks for flagging this. Is it possible to find a workaround? Can we check that and let the users know that such an attribute already exists? It'd be handy for both attributes and options (f.e. for existing users who want to create variations using attributes, instead of options).

cc @pmcpinto as this is a limitation I don't think we were aware of.

**Comment by louwie17** [comment]
@jarekmorawski @mattsherman just chiming in here as this conversation is moving along. I am just reading through the comments and almost getting confused myself on when we should use which. And if we developers are getting confused, I get a feeling that merchants could be more confused, **especially** if they are used to the old UI.

I am actually curious if the separation between attributes and options is causing more harm then simplicity.
I know we already settled on the UI for variations, but I wonder if we are better off not really differentiating between the two, meaning we still show all global/local attributes in both spots, we just make abundantly clear whether the specific attribute is used as an option (for variation) or an attribute.

If used as an option, we can add a link on the attributes field (edit in options) and vice versa.

> > There is no variation flag on the global attributes themselves.
>
> We might want to add it at some point in the future, but only if and when we decide to create a global Products > Options screen.

If we did go this route, I think we should remove the **use for variations** flag as a whole and keep attributes only as attributes and options only for variations. That way users can create an attribute and an option with the same name and use them both within a variable product and a simple product (respectively).

**Comment by jarekmorawski** [comment]
Thanks for sharing your thoughts, @louwie17. You raised a question I've asked myself many times in the last few months. However, no user research would suggest that people find the new structure confusing. On the contrary, based on the collected feedback, they like having both options and variations in one place.

> I get a feeling that merchants could be more confused, especially if they are used to the old UI.

It is a real risk, but we hope to mitigate it by educating users about the new structure. We hope it won't take the users too long to adjust their thinking, especially when the new experience is simpler and faster.

What exactly confuses you about the new setup? You wouldn't know whether, say, `Color` should be an attribute or option?

> we still show all global/local attributes in both spots, we just make abundantly clear whether the specific attribute is used as an option (for variation) or an attribute.

I see your point. However, this would create complexity because users would manage the two attribute types in different places in the UI. It would also not be crystal clear where they could edit an attribute used as an option. In Options or Attributes? What if we introduce different UI layouts for options (f.e. color swatches or radio buttons)? Should those settings be managed in Options while others in Attributes?

> I think we should remove the use for variations flag as a whole and keep attributes only as attributes and options only for variations.

That's exactly the route we're taking. Sorry if it wasn't clear. There will be no variation flag to enable in the new UX. It will only exist on the backend to ensure backward compatibility. Once we roll out the new UI, all attributes created in Options will have that flag enabled, and users will not be able to disable it in any way. Thinking ahead, we'd introduce options as separate database objects, but not now, given how important it is to maintain compatibility with the existing structure.

**Comment by mattsherman** [comment]
> how will the user sort all the attributes in a single ordering?
> > How are they sorted now? They'll be two separate lists, so we can use the existing sorting system.

They are sorted in a single list currently.

<img width="808" alt="Screenshot 2022-12-09 at 08 37 14" src="https://user-images.githubusercontent.com/2098816/206714499-cf3e32d8-aefb-491f-a126-07469313ec88.png">

If we let users separately sort the items in each list, how do we arrive at an overall sorting for the combined set of attributes that makes sense to the user?

**Comment by louwie17** [comment]
> On the contrary, based on the collected feedback, they like having both options and variations in one place.

Yeah I do certainly like that design and not opposed to that.

> What exactly confuses you about the new setup? You wouldn't know whether, say, Color should be an attribute or option?

I think the base of the confusion is mostly around the implementation, of hiding the the options within attributes and vice versa. Also the change that users could create the same attribute and option and how we would differentiate between the two. I think the latter could potentially be confusing for the user to.

Given we will only support custom options in the variations view should help with this. Although I do wonder how that will go with backwards compatibility, for users that have enabled variations for a global attribute already (for the specific product).

> It would also not be crystal clear where they could edit an attribute used as an option. In Options or Attributes? What if we introduce different UI layouts for options (f.e. color swatches or radio buttons)? Should those settings be managed in Options while others in Attributes?

All good points/questions. Also in relation to what Matt said, if we remove an option, do we remove this from the attributes list as well or keep with with the `used for variations` set to false.
The current layout/design seems like a good long term approach with the goal of creating an options table/data type, specifically for variations.
Only supporting custom attributes as options within variations does seem like a good initial step in that direction. I am mostly curious if there are one or two other things we can do to bridge that gap for existing products made with the old UI.

I wonder if users edit an existing variation product with global attributes/options, that we show a notice to automatically turn these into custom options for now. Of-course this would quickly become confusing if the user goes back to the old UI and looks at the attributes screen.

> That's exactly the route we're taking. Sorry if it wasn't clear. There will be no variation flag to enable in the new UX. It will only exist on the backend to ensure backward compatibility. Once we roll out the new UI, all attributes created in Options will have that flag enabled, and users will not be able to disable it in any way. Thinking ahead, we'd introduce options as separate database objects, but not now, given how important it is to maintain compatibility with the existing structure.

👍 👍 

> Oh, really! Thanks for flagging this. Is it possible to find a workaround? Can we check that and let the users know that such an attribute already exists? It'd be handy for both attributes and options (f.e. for existing users who want to create variations using attributes, instead of options).

I believe the only workaround right now is to modify the slug, to always prepend/append something like `wcoption-`. As we can have attributes with the same name, the slug just has to be unique. I don't think the slug shows up in to many places, aside from maybe in the url if the user filters by attribute.

**Comment by jarekmorawski** [comment]
> Also the change that users could create the same attribute and option and how we would differentiate between the two. I think the latter could potentially be confusing for the user to.

We won't prevent all possible mistakes, but we can proactively educate users. The new experience is a major change and will come with a mild learning curve–there's no way around it. Merchants can create Color as both an attribute and an option, and I'm certain that when they see that they're displayed differently on the product page, they'll understand the difference and learn what's used when, and how (hopefully 🤞).

> Also in relation to what Matt said, if we remove an option, do we remove this from the attributes list as well or keep with with the used for variations set to false.

It won't show up in the attributes list in the first place, so there'd be nothing to hide. If you remove an option, you'd no longer see it in the Options list, and the variations table would update. It'd have no impact on added attributes whatsoever. Attributes and Options are not connected on the front-end anymore.

> I am mostly curious if there are one or two other things we can do to bridge that gap for existing products made with the old UI.

Certainly. For example, we'd detect that the user added an option with the same name as an attribute and let them remove the duplicate. We will likely come up with more ideas once we start testing the new UX internally and externally.

> I wonder if users edit an existing variation product with global attributes/options, that we show a notice to automatically turn these into custom options for now.

Possibly. Global attributes get very little use, so we treat them as an edge case. All attributes with the variation flag enabled for existing products should be moved to the Options tab–and if they're global, we should make local copies.

**Comment by pmcpinto** [comment]
Hey everyone, thanks for your feedback. We're trying to simplify the experience for merchants with a complex foundation. One way or another, we'll need to make some tradeoffs to ensure backward compatibility. Let me think about this over the weekend

**Comment by mattsherman** [comment]
> Merchants can create Color as both an attribute and an option

I don't think we should allow this. This currently is prevented in the existing UI and would be confusing outside of the new product experience UI.

> Attributes and Options are not connected on the front-end anymore.

This is not really true. They still affect each other, even though we show them separately.

> we'd detect that the user added an option with the same name as an attribute and let them remove the duplicate

Is this is a future possibility or part of this epic?

> Global attributes get very little use, so we treat them as an edge case

Do we have metrics for this?

> All attributes with the variation flag enabled for existing products should be moved to the Options tab–and if they're global, we should make local copies.

Making local copies without informing the user would be confusing and unexpected. I think we should be very transparent what is being done.

And, I'm not sure this is the ideal approach. Personally, I think that we should continue to support global attributes until we deprecate and remove them from WooCommerce altogether.

**Comment by mattsherman** [comment]
> I am actually curious if the separation between attributes and options is causing more harm then simplicity.
I know we already settled on the UI for variations, but I wonder if we are better off not really differentiating between the two, meaning we still show all global/local attributes in both spots, we just make abundantly clear whether the specific attribute is used as an option (for variation) or an attribute.
> If used as an option, we can add a link on the attributes field (edit in options) and vice versa.

@louwie17 I lean that way as well, as an initial iteration.

**Comment by mattsherman** [comment]
> We're trying to simplify the experience for merchants with a complex foundation. One way or another, we'll need to make some tradeoffs to ensure backward compatibility.

I agree that ensuring backward compatibility is very important. Our new product screen will not be the only means of viewing and modifying products created through it. We need to consider what will happen when these products and viewed and modified through the existing product screen and the APIs.

There are a few open questions/issues that we need to resolve:

- Sorting of Options and Attributes. If we allow the user to separately sort the Options and Attributes in the new UI, we don't have a clear combined sort order, as exists in the existing UI.
- Are we continuing to support global Options? If we do not, how do we maintain backwards compatibility?
- Will we support having both an Option and an Attribute with the same name? This is prevented in the current UI, though I don't believe it is technically prevented in the underlying API or DB (need to double-check this). If we do, I think this will result in a confusing experience, both on the store-front and in the existing UI and API.

I personally think that a user should be able to go back and forth between the new and existing product screen to edit a product. If for whatever reason, whether to use an extension that isn't yet supported in the new screen, or just because the user wants to go back to a more familiar experience for a bit to get something done quickly, there should be no concern that editing a product created in one screen in the other screen will have any issues.

We also need to ensure that products created via the new screen can be modified via the API and vice-versa without issue.

**Comment by jarekmorawski** [comment]
> I don't think we should allow this. This currently is prevented in the existing UI and would be confusing outside of the new product experience UI.

Unless you think we can build a bulletproof logic that would detect and prevent users from creating duplicates, I'd say this is something we can explore at a later stage. Once people understand the difference between attributes and options, it's very unlikely they'll try to use both for the same purpose.

> This is not really true. They still affect each other, even though we show them separately.

Not to the user. I'm all ears if you think there's a better way to structure things on the API/DB level. UX-wise, we should keep attributes and options separate because the current relationship between the two causes much confusion.

> Is this is a future possibility or part of this epic?

Future possibility. It's likely a bigger problem to solve than a simple diff check.

> Do we have metrics for this?

Yes. We also track how often merchants add attributes while creating and updating products. Almost all products are published without attributes, and a low % of merchants at no more than 1 attribute at a later time.

> And, I'm not sure this is the ideal approach. Personally, I think that we should continue to support global attributes until we deprecate and remove them from WooCommerce altogether.

> We need to consider what will happen when these products and viewed and modified through the existing product screen and the APIs.

Good point. Can you please take some time to identify and list all possible edge cases and limitations? It'll make it easier for us to understand the problem from the developer's POV and update the flows.

I'd also appreciate it if you could think about what would it take for us to build the new UX the way we envisioned it. Maybe adding a new DB prop? A separate DB for products edited using the new UX? A migration tool? I'd like us to think of solutions, even if we decide to keep the current structure temporarily.

> Sorting of Options and Attributes. If we allow the user to separately sort the Options and Attributes in the new UI, we don't have a clear combined sort order, as exists in the existing UI.

What sorting do you mean? Regular and variation-enabled attributes are displayed in different places on the product page, so they don't have a combined sort order. What is the sort order used for?

> Are we continuing to support global Options? If we do not, how do we maintain backwards compatibility?

Global options, as in global attributes with the variation flag enabled? If yes, I don't think they exist in the current system, do they? You can have global attributes and enable the flag on a single product level, but not for all products where they're added.

If a product has global attributes with the variation flag enabled, we will show them in Options. What do you think could be a blocker?

> If we do, I think this will result in a confusing experience, both on the store-front and in the existing UI and API.

I agree, and as we said above, we could work out an elegant solution to prevent or at least communicate that clearly to the merchants.

> I personally think that a user should be able to go back and forth between the new and existing product screen to edit a product.

100% agree. We will let users easily switch back and forth for at least a few months after the new UX is released. I wonder if we can create workarounds that would bridge the data structure between the current and new attributes. Otherwise, the new form will only be a fresh coat of paint on top of a lacking, confusing experience.

> We also need to ensure that products created via the new screen can be modified via the API and vice-versa without issue.

I am curious to learn more about this use case. How can users create products via the API? Is it related to extensibility? Can we update the API without breaking compatibility? I'm happy to discuss this elsewhere since it's not 100% related to this epic.

**Comment by pmcpinto** [comment]
@mattsherman @louwie17 this thread is getting long. Can you summarize your feedback and add it to the post published by Jarek 2 weeks ago, where he shared this idea? pbIJXs-2Yx-p2
Let's define the next steps and potentially some design refinements in the post. Later, I or Jarek will adjust the acceptance criteria of this epic.
cc @AnnaMag

**Comment by louwie17** [comment]
That sounds like a good plan @pmcpinto 👍 
@mattsherman I will ping you to see if we can sync up today or tomorrow and summarize our thoughts (depending on how much they overlap :) otherwise we can post two separate P2 comments)

**Comment by joshuatf** [comment]
Thanks for the discussion here @mattsherman and @louwie17.

For anything related to the design and confusion around options vs attributes, I think we should take that to p2 so we can keep our GH issues a bit more implementation-focused.

As for technical challenges, I think there are some good comments here.  Wherever possible, we should move these discussions to the individual issues above to keep discussion length down and visibility higher.

>Sorting of Options and Attributes. If we allow the user to separately sort the Options and Attributes in the new UI, we don't have a clear combined sort order, as exists in the existing UI.

Good question.  I wonder if there are any areas where both attribute and options can be seen together in a customer-facing area.  If so, this would require rethinking design a bit.  If not, this is really more of a technical challenge for us to solve and we can probably sort each separately even if the sorting ultimately saves to the same place in the DB.

>Making local copies without informing the user would be confusing and unexpected. I think we should be very transparent what is being done.

💯 

>Will we support having both an Option and an Attribute with the same name? This is prevented in the current UI, though I don't believe it is technically prevented in the underlying API or DB (need to double-check this). If we do, I think this will result in a confusing experience, both on the store-front and in the existing UI and API.

Great point.  I think this is a good question, even if it's an edge case.  For the sake of simplicity in this cycle, I'd vote to hold off on adding this logic and we can revisit next cycle.

**Comment by mattsherman** [comment]
I've summarized @louwie17 and my thoughts here: pbIJXs-2Yx-p2#comment-5725

This epic, and other issues, will be updated accordingly after things are discussed further on that post.

**Comment by pmcpinto** [comment]
@joshuatf @jarekmorawski the AC was updated based on the P2 and Google Doc shared by Jarek. Pls take a look at it, as it's a long list covering interdependent use cases so I may have missed something. Thanks!

**Comment by pmcpinto** [comment]
> Each product option should be saved as a global attribute used for variations. By default, all attributes are created globally. We inform the user about this in the add attributes modal.

One thing to keep in mind is the global attribute scalability mentioned [here](https://github.com/woocommerce/woocommerce/issues/20262#issuecomment-401313878), as we explore the idea of having all new attributes set as global by default. cc @joshuatf @jarekmorawski

**Comment by joshuatf** [comment]
@pmcpinto Thank you for the updates!  

I think going forward we might want to create new issues or a new epic with the changes.  Added it here makes it a little challenging to see what changes might have been made and creates a bit of a moving target on WIP.

**Comment by pmcpinto** [comment]
@joshuatf Good point. I can create a new epic with the missing changes after I review the work that was merged and included in the test store. Sounds good?

**Comment by pmcpinto** [comment]
I'll close this epic as it was created a new one with the missing pieces: #36286
